dashboard/src/components/SideMenu/SideMenu.tsx

```diff
@@ -64,15 +64,12 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
 
   const isCurrentPath = pathname.startsWith(item.navigateTo);
 
-  const navigate = (): void => {
-    const origin = zOrigin.parse(unsafeOrigin);
-    const finalPath = item.navigateTo;
-    const newPath = `${finalPath}?origin=${origin}`;
+  const paths = ['/tree', '/devices', '/labs'] as const;
+
+  const zPathEnum = z.enum(paths);
+  const zPath = zPathEnum.catch('/tree');
 
-    router.navigate({
-      to: newPath,
-    });
-  };
+  const safeNavigateTo = zPath.parse(item.navigateTo);
```

Comment:
Actually since item is defined in code level, it doesn't even need zod (it doesn't word to leave it here), this could be just the item prop type definition